### ISSUE TITLE
Server multi-port / client port-hopping

### ISSUE THREAD

**Issue by tobyxdd** [HyNetwork/hysteria#480]
Listen address/server address should support the following format:

`server.com:1234,5678,6969`

This indicates that the server is available on ports 1234, 5678, and 6969.

The client can choose any of these ports to connect, and randomly hop to a different one every once in a while. **The server must save the mapping between client addresses & server ports and respond to the client using the client's selected port (to avoid NAT issues).**

Ideally, the client should also use a different local port for each server port, making both the source and destination ports distinct with each hop.

**Comment by ddk2222-luck** [comment]
这样不错，一来避免ISP的QOS，二来长时间链接一个境外UDP也很容易被墙。

**Comment by tobyxdd** [comment]
> How about ports range? `server.com:1234,5678-6969`

I'm not exactly keen on adding range support. It would be easy for users to specify huge ranges, making their servers listen on tens of thousands of ports without realizing the performance and resource implications (each socket has its buffers and receive routine).

That said, you can still listen on many ports by using many commas with the current syntax. There will be no artificial limit.

**Comment by tobyxdd** [comment]
> How about ports range? `server.com:1234,5678-6969`

After some internal discussions, we decided to add port range support to the client anyway.

However, we removed the native multi-port support on the server - the server will only listen on one port, as in the current release. Users must use something like iptables DNAT / netsh portproxy to handle port ranges themselves. This is much better in terms of performance. The only downside is that users will need to figure out how to configure it properly by themselves.

2e803348419dd9853c4fe5881c9cd668ccce5476
